### DISCUSSION TITLE
`extendr` or `savvy`?

### DISCUSSION THREAD

**Discussion by asbates** [r-staceans/blazr#6] [Q&A] [answered]
There are currently 2 ways to use Rust in an R package: [`extendr`](https://github.com/extendr/extendr) and [`savvy`](https://github.com/yutannihilation/savvy).

I think `extendr` is the more established framework while `savvy` is simpler. Other than that I'm not that familiar with the differences. Personally I tend to favor simpler options first so I suggest we consider `savvy`. If we find that `savvy` doesn't meet our needs we can always convert to `extendr` later.

@jdhoffa @jibarozzo what do you think?

**Comment by asbates** [comment]
Apparently the author of the R wrapper to [`polars`](https://docs.rs/polars/latest/polars/) decided to rewrite the whole package into [`neopolars`](https://github.com/eitsupi/neo-r-polars?tab=readme-ov-file). Part of the reason for this is related to issues with `extendr` so they switched to `savvy` for the new version.

**Comment by jibarozzo** [comment]
Interesting. Simple sounds good. The main issues seem to be with the feature-rich environment of `extender` . https://yutannihilation.github.io/savvy/guide/extendr.html

**Comment by jdhoffa** [comment, marked as answer]
Thanks to @jibarozzo for a very succinct summary of the differences here:
https://github.com/r-staceans/multiverse/pull/7#issuecomment-2525395953

I think for the purposes of this repository, `savvy` makes the most sense. 

Recommending to close this discussion as decided

**Reply by jdhoffa** [reply]
@asbates @jibarozzo feel free to re-open if you think it warrants further investigation!
